### DISCUSSION TITLE
Possible error in calculating spaces to add during tab-to-space munging?

### DISCUSSION THREAD

**Discussion by vsajip** [javacc21/javacc21#199] [Q&A] [answered]
I think in the following line:
https://github.com/javacc21/javacc21/blob/7bce8c1c101a57d2581094e53a492aa08e0bb2a7/src/ftl/java/Lexer.java.ftl#L749
we should be using the `tabSize` lexer field instead, as initialized here but sometimes settable to a different value:
https://github.com/javacc21/javacc21/blob/7bce8c1c101a57d2581094e53a492aa08e0bb2a7/src/ftl/java/Lexer.java.ftl#L81-L85
Can you please confirm, @revusky ?

**Comment by revusky** [comment, marked as answer]
I don't think there's any problem with the code you're pointing to -- well, aside from the possibility that it is written in a rather confusing way! 

Now, in principle, it is only possible to set tabSize to something other than DEFAULT_TAB_SIZE if PRESERVE_TABS is true. That's why line 84 above is only there if grammar.preserveTabs is true. So, if we're not preserving tabs, i.e. they are all getting converted to spaces on initialization, then the other bit of code, line 749, it doesn't really matter if we use DEFAULT_TAB_SIZE or tabSize, because it's the same anyway. (Yeah, I know it's expressed in a rather confusing way, but that is surely an artifact of how this whole thing evolved...)

But, anyway, line 749 is only active if preserveTabs is false, which means that we're converting all the tabs to spaces, and it also means that, effectively, the issue is pretty much moot after the content is slurped in because we don't have any tabs any more anyway! You see, the place where tabSize is used is when we're figuring out column locations. That's line 592, but that line is actually dead code if preserveTabs is false, because we don't have any tab characters anyway!

Well, anyway, I'm pretty sure there's not any problem with this code, as regards it working correctly, but there could be improvement in terms of it being expressed more clearly. It is a bit confusing the way it is now, proof of that being that you found it confusing! (I would probably find it confusing too if I hadn't written it myself!)
